includes/collection/class-interactions.php

```diff
@@ -34,10 +34,19 @@ class Interactions {
 	public static function add_comment( $activity ) {
 		$comment_data = self::activity_to_comment( $activity );
 
-		if ( ! $comment_data || ! isset( $activity['object']['inReplyTo'] ) ) {
+		if ( ! $comment_data ) {
 			return false;
 		}
 
+		if ( empty( $activity['object']['inReplyTo'] ) ) {
+			$activity = self::extract_quote_link( $activity );
```

Comment:
Let's return the URL of the quoted post here and pass it directly to `\url_to_postid()`, since that's what we're after, the comment_post_ID.